src/meta/client/MetaClient.cpp

```diff
@@ -643,6 +715,126 @@ MetaClient::removeRange(std::string segment, std::string start, std::string end)
                 }, true);
 }
 
+folly::Future<StatusOr<UserID>>
+MetaClient::createUser(cpp2::UserItem userItem, std::string password, bool missingOk) {
+    cpp2::CreateUserReq req;
+    req.set_user(std::move(userItem));
+    req.set_encoded_pwd(std::move(password));
+    req.set_missing_ok(missingOk);
+    return getResponse(std::move(req), [] (auto client, auto request) {
+        return client->future_createUser(request);
+    }, [] (cpp2::ExecResp&& resp) -> UserID {
+        return resp.get_id().get_user_id();
+    }, true);
+}
+
+folly::Future<StatusOr<bool>>
+MetaClient::dropUser(std::string account, bool missingOk) {
+    cpp2::DropUserReq req;
+    req.set_account(std::move(account));
+    req.set_missing_ok(missingOk);
+    return getResponse(std::move(req), [] (auto client, auto request) {
+        return client->future_dropUser(request);
+    }, [] (cpp2::ExecResp&& resp) -> bool {
+        return resp.code == cpp2::ErrorCode::SUCCEEDED;
+    }, true);
+}
+
+folly::Future<StatusOr<bool>>
+MetaClient::alterUser(cpp2::UserItem userItem) {
+    cpp2::AlterUserReq req;
+    req.set_user_item(std::move(userItem));
+    return getResponse(std::move(req), [] (auto client, auto request) {
+        return client->future_alterUser(request);
+    }, [] (cpp2::ExecResp&& resp) -> bool {
+        return resp.code == cpp2::ErrorCode::SUCCEEDED;
+    }, true);
+}
+
+folly::Future<StatusOr<bool>>
+MetaClient::grantToUser(cpp2::RoleItem roleItem) {
+    cpp2::GrantRoleReq req;
+    req.set_role_item(std::move(roleItem));
+    return getResponse(std::move(req), [] (auto client, auto request) {
+        return client->future_grantRole(request);
+    }, [] (cpp2::ExecResp&& resp) -> bool {
+        return resp.code == cpp2::ErrorCode::SUCCEEDED;
+    }, true);
+}
+
+folly::Future<StatusOr<bool>>
+MetaClient::revokeFromUser(cpp2::RoleItem roleItem) {
+    cpp2::RevokeRoleReq req;
+    req.set_role_item(std::move(roleItem));
+    return getResponse(std::move(req), [] (auto client, auto request) {
+        return client->future_revokeRole(request);
+    }, [] (cpp2::ExecResp&& resp) -> bool {
+        return resp.code == cpp2::ErrorCode::SUCCEEDED;
+    }, true);
+}
+
+folly::Future<StatusOr<cpp2::UserItem>>
+MetaClient::getUser(std::string account) {
+    cpp2::GetUserReq req;
+    req.set_account(account);
+    return getResponse(std::move(req), [] (auto client, auto request) {
+        return client->future_getUser(request);
+    }, [] (cpp2::GetUserResp&& resp) -> cpp2::UserItem {
+        return std::move(resp).get_user_item();
+    });
+}
+
+folly::Future<StatusOr<std::unordered_map<UserID, cpp2::UserItem>>>
+MetaClient::listUsers() {
+    cpp2::ListUsersReq req;
+    return getResponse(std::move(req), [] (auto client, auto request) {
+        return client->future_listUsers(request);
+    }, [] (cpp2::ListUsersResp&& resp) -> decltype(auto) {
```

Comment:
Please do a research on the difference between `auto` versus `decltype(auto)`, don't use the latter one unless it's necessary.